### PR TITLE
Add optional mixed types syntactic sugar

### PR DESCRIPTION
**Pull request by mratsim**
Solve #15 

Introduce optional mixed types syntactic sugar.

The goal is to keep a natural syntax as much as possible and avoid having conversion everywhere when working with mixed uint256, int and uint in a codebase for example.

The file must be explicitly imported with `import stint/mixed_types_ops` so that when strong static typing is preferred or for debugging, conversions are required to be explicit.

### CODE REVIEW

**Comment by yglukhov** [comment]
"Optionality" defeats the purpose, IMO :)

**Comment by zah** [comment]
Is it too complicated to provide true specializations, avoiding the zero-initialization of the large stint values?

**Comment by mratsim** [comment]
@zah do you mean true specialization for `a == 10`?

It can be done, by iterating on the most significant words make sure, those are 0 and only compare the least significant word. I have to introduce a new macro similar to asWordsZip: https://github.com/status-im/nim-stint/blob/8e25b30209e951bcabfc2d1cd97ef0b971554648/stint/private/as_words.nim#L116

@yglukhov 

The purpose of static typing is also avoiding that ;) :
```Nim
let a: int64 = 1
let b: uint64 = 1
echo a + b
```

**Comment by mratsim** [comment]
Naming: should it be called `lenient_stint` in the spirit of `lenientops`? https://nim-lang.org/docs/lenientops.html

**Comment by mratsim** [comment]
@zah, I've been thinking over the night about the implication of specialization.

It is not complicated but it requires specific handling for most:
- specific handling for addition for carry
- specific handling for substraction for borrow
- specific handling for multiplication carries
- Division/mod I'm not sure, I don't think we can specialize, we need to shift the most significant bit of the divisor to the MSB of the dividend to do fast division.

- xor/or/and with the asWordsZip-like macro I talk about in my previous post.

So in short it's reimplementing most of the library and this will require new tests. It can probably wait or even be put as a bounty.
